util/env_posix.cc

```diff
@@ -216,30 +229,15 @@ class PosixEnv : public Env {
         }
       }
       close(fd);
```

Comment:
Oh, it works because PosixMmapReadableFile stores, but doesn't use, its fd_ instance variable. Let's get rid of it. There's even a hack "fd_ = fd_ + 0;  // suppress the warning for used variables" to suppress a very correct compiler warning, which we'll also be able to get rid of.